src/components/input/input.js

```diff
@@ -650,7 +651,21 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       // Using the $validators for triggering the update works very well.
       renderCharCount();
```

Comment:
It might be preferrable to call [ngModelCtrl.$validate()][1] instead, since that will take care of updating input/form `$error`s, validation classes, etc

[1]: https://code.angularjs.org/1.6.9/docs/api/ng/type/ngModel.NgModelController#$validate